src/hooks/useBodyLocked.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useLayoutEffect, useState } from "react";
```

Comment:
Awesome hook composition!

Reply:
`useLayoutEffect` was a new one for me too.